vignettes/stan-help.Rmd

```diff
@@ -0,0 +1,423 @@
+
+### `chains`
+
+One of the key components of Bayesian inference is creating draws from independent Markov chains (See [this post](https://betanalpha.github.io/assets/case_studies/markov_chain_monte_carlo.html#2_markov_chain_of_command) by Michael Betancourt for a more detailed introduction).
+To assess convergence and the reliability of a given posterior distribution it is vital that at least two chains are used with **`4` chains** being the more common default.
+
+### `threads_per_chain`
+
+Stan allows for multi-threading and the Stan code used within `{epinowcast}` has tried to take advantage of these capabilities where possible.
+Multi-threading allows for the calculations to be spread across multiple threads, which could accelerate inference by allowing embarrassingly parallel calculations to be spread across threads and then later combined.
+If possible, a good default is `2`, however, if you have available compute on your machine, you can increase this to a higher value and may see some fit time reductions.
+Please note that the `theads_per_chain` has not impact on model convergence or fittings and is purely a means of speeding up the model fitting time.
+
+::: {.alert .alert-warning}
+**The product of the `thread_per_chain` and `chains`** arguments should be equal to or less than the number of cores on your machine! 
+:::
+
+### iter_warmup and iter_sampling {.iters}
+
+The number of samples using for the warmup phase of Stan and the number of sampling iterations are controlled by the `iter_warmup` and `iter_sampling` arguments, respectively.
+Stan requires a sufficient number of warmup iterations to dynamically estimate the step change sizes during model fitting. 
+Unlike with Gibbs sampling (such as the method used in JAGS), you generally do not need to set large values for `iter_warmup` or `iter_sampling`, but this can vary based on the [match between your data and your model](https://discourse.mc-stan.org/t/number-of-iterations/1674/3).
+Conversely, setting large values for iterations could be a sign of poor model fit and indicates that other approaches need to be employed (e.g., `priors`, `adapt_delta`, `max_treedepth`).
+A [general heuristic](https://mc-stan.org/rstan/reference/Rhat.html#:~:text=Both%20bulk%2DESS%20and%20tail,respective%20posterior%20quantiles%20are%20reliable) is to have your bulk and tail effective sample sizes (ESS) greater than 100.
+However, [depending on the complexity of your problem](https://discourse.mc-stan.org/t/number-of-iterations/1674/13) increasing the number of iterations can be appropriate (i.e., your [$\hatR$ values are near 1](https://arxiv.org/pdf/1903.08008.pdf), [your Monte Carlo Standard errors](https://mc-stan.org/docs/cmdstan-guide/stansummary.html) are low) especially if you want to make inferences regarding tail behaviours.
+Typically we have found that 1000-2000 samples per chain for `iter_warmup` and `iter_sampling` are good starting values for initial model fitting.
```

Comment:
This is true in terms of the stan docs but I have found that generally can get away with fewer warmup samples for many models - we should perhaps mention this a bit as it will really speed up peoples work

Reply:
Very true. What's your default for warm up? 500-750?

Reply:
250 (I am a terrible human) but I think 500 is generally pretty safe.